File: libraries/dotnet/logs/src/ExporterParameters.cs
// This file contains exporter parameters logic for logs src.
using System.Text.Json.Serialization;

namespace CloudOps.Otel.Logs;

public sealed class ExporterParameters
{
    [JsonPropertyName("otel")]
    public BackendConfig? Otel { get; set; }

    // Checks whether empty.
    public bool IsEmpty()
    {
        return Otel?.Logs == null
            || (string.IsNullOrWhiteSpace(Otel.Logs.Url) && string.IsNullOrWhiteSpace(Otel.Logs.ApiKey));
    }

    // Handles backend.
    public BackendConfig? Backend(string name)
    {
        return name == "otel" ? Otel : null;
    }
}

File: libraries/java/logs/src/main/java/otel/logs/ExporterParameters.java
// This file contains exporter parameters logic for OTel logs.
package otel.logs;

import java.util.Objects;

public final class ExporterParameters {
  public BackendConfig otel;

  // Checks whether empty.
  public boolean isEmpty() {
    return otel == null || otel.logs == null || !hasValue(otel.logs.url);
  }

  // Handles has value.
  private static boolean hasValue(String value) {
    return value != null && !value.isBlank();
  }

  // Handles backend.
  public BackendConfig backend(String name) {
    if (Objects.equals(name, "otel")) return otel;
    return null;
  }
}
